apps/src/applab/dataBrowser/EditTableRow.jsx

```diff
@@ -1,35 +1,100 @@
+import FirebaseStorage from '../firebaseStorage';
 import Radium from 'radium';
 import React from 'react';
+import { castValue, displayValue } from './dataUtils';
 import * as dataStyles from './dataStyles';
 
 const EditTableRow = React.createClass({
   propTypes: {
     columnNames: React.PropTypes.array.isRequired,
+    tableName: React.PropTypes.string.isRequired,
     record: React.PropTypes.object.isRequired
   },
 
+  getInitialState() {
+    return {
+      isEditing: false
```

Comment:
OK, I've changed it to initialize `newRecord: {}` because https://facebook.github.io/react/tips/props-in-getInitialState-as-anti-pattern.html. also `newValue: undefined` in EditKeyRow.jsx